### PR TITLE
Merge args into config

### PR DESCRIPTION
**Pull request by epage**
This is preparation for #93.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -7,55 +7,349 @@ use serde::{Deserialize, Serialize};
...
-            tag_prefix: None,
-            doc_commit_message: "(cargo-release) generate docs".into(),
-            disable_tag: false,
-            enable_features: vec![],
-            enable_all_features: false,
-            dependent_version: Default::default(),
+impl Config {
+    pub fn update(&mut self, source: &ConfigSource) {
+        if let Some(sign_commit) = source.sign_commit() {
+            self.sign_commit = Some(sign_commit);
```
Comment (sunng87) [inline src/config.rs]: why not just `self.sign_commit = source.sign_commit().clone()`

Comment (epage) [reply src/config.rs]: Either way. Another option is to encapsulate this pattern with an extension trait.

Comment (epage) [reply src/config.rs]: Thinking more about this...

The current approach references the struct twice with the proposed referencing it three times which increases the chance of error.

Comment (sunng87) [reply src/config.rs]: Or is it possible to write a procedural macro for this? To avoid multiples changes required for adding new config option.

**Comment by sunng87** [comment]
@epage do you want to improve on this branch, or I will merge it to let the process go

File: src/config.rs (hunk 1 above)
Comment (epage) [reply src/config.rs]: Yes, a procedural macro could help here.  Maybe I'm just not familiar enough with them but it seems like a big hammer to solve this small problem. 

I do want to take these lessons back to the CLI-WG [discussions](https://github.com/rust-lang-nursery/cli-wg/issues/7).  It feels like the current selection of crates is too all-in-one which doesn't let us adapt for the needs of crates like this one.  I'm thinking a generalized set of macros to let people layer their config as needed would be helpful for the community.

**Comment by epage** [comment]
> @epage do you want to improve on this branch, or I will merge it to let the process go

If you are fine with it, I'd say let's get this in and continue to iterate.

**Comment by sunng87** [comment]
Yes.

File: src/config.rs (hunk 1 above)
Comment (sunng87) [reply src/config.rs]: I had a [clojure library](https://github.com/sunng87/stavka), which I think deals this well. I'll think about the challenge to port it to static-typed language like rust.

Comment (epage) [reply src/config.rs]: `config` and `configure` do it to a degree but I think they both use a `HashMap` when merging the sources which delays error reporting to the end.  My goal with this was to report the error (invalid data type, extra field, etc) with being able to tell the user where the problem came from so they could fix it.

**Pull request merged**